### DISCUSSION TITLE
shm transport delay is too large

### DISCUSSION THREAD

**Discussion by zonghai-yang** [eProsima/Fast-DDS#5463] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I hope the latency of writing big data is high and the latency of writing small data is low.

### Current behavior

I use two processes.Process 1 has two datawriters.Process 2 has two datareaders.The topic of datawriter1 and datareader1 is topic1.
The topic of datawriter2 and datareader2 is topic2.Datawriter1 write a large size msg1.Datawriter2 write a small size msg2.I hope delay of msg1 is long and delay of msg2 is small.But delay of msg2 also long.I think it was influenced by writer1.I changed the sending order of writer1 and writer2 and sent small data first. The latency of small data has become lower。

### Steps to reproduce

process_1:
sizeof(msg_1) == 1M;
sizeof(msg_2) == 10byte;
datawriter_1.topic() = "topic_1";
datawriter_2.topic() = "topic_2";
datawriter_1.write(msg_1);
datawriter_2.write(msg_2);
//if write msg_2 first , delay will correct

process_2:
datareader_1.topic() = "topic_1";
datareader_2.topic() = "topic_2";
std::cout<<now - msg_1.timestamp();
std::cout<<now - msg_2.timestamp();

### Fast DDS version/commit

v2.11.2

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

**Comment by cferreiragonz** [comment]
Hi @zonghai-yang,

Thank you for using Fast DDS!

In Fast DDS, transports are configured at the Participant level. This means that two DataWriters instantiated within the same Participant will share the same transports (UDP and SHM in your case). Consequently, any delay introduced by one DataWriter may impact the others. However, there are two approaches you can try to mitigate this effect:

- Use [Flow Controllers](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/large_data/large_data.html#flow-controllers):
    Flow Controllers provide a mechanism to limit the rate at which data is sent by a DataWriter. Additionally, they allow you to configure scheduling policies, enabling you to assign a higher priority to the second writer or reserve specific bandwidth for it. 

- Optimize for large message latency:
    Reducing the latency experienced when sending large messages can solve your problem. I highly recommend reviewing our [guidelines for working with large data rates](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/large_data/large_data.html#large-data-rates). These include strategies such as enabling the [LARGE_DATA](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/tcp/tcp_with_multicast_discovery.html#large-data-mode) mode or [adjusting socket buffer sizes](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/large_data/large_data.html#increasing-socket-buffers-size).

I also encourage you to upgrade to a maintained Fast DDS version like [2.14](https://github.com/eProsima/Fast-DDS/releases/tag/v2.14.4) or [3.1](https://github.com/eProsima/Fast-DDS/releases/tag/v3.1.0)!

Lastly, as this question is more related to support rather than a software issue, I am moving this thread to its corresponding discussion forum in accordance with the [Fast DDS contributing guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support).

If you have further questions, feel free to reach out!
